src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -2689,6 +2690,15 @@ public DocumentFlags GetFlagsFromOldDocument(DocumentFlags newFlags, DocumentFla
             return newFlags;
         }
 
+        public DocumentFlags GetFlagsFromOldDocumentForDelete(DocumentFlags newFlags, DocumentFlags oldFlags, NonPersistentDocumentFlags nonPersistentFlags)
+        {
+            var flags = GetFlagsFromOldDocumentForPut(newFlags, oldFlags, nonPersistentFlags);
+            if (oldFlags.Contain(DocumentFlags.Artificial))
+                flags |= DocumentFlags.Artificial;
```

Comment:
Assuming that we do not need `FromIndex` flag?

Reply:
yep, I don't think we do...